mlir/docs/DefiningDialects/Operations.md

```diff
@@ -1026,6 +1055,8 @@ to:
 1.  All operand and result types must appear within the format using the various
     `type` directives, either individually or with the `operands` or `results`
     directives.
+1.  Unless all non-attribute properties appear in the format, the `prop-dict`
+    directive must be present.
```

Comment:
I'm actually working on a patch to go further in the proper separation of discardable and inherent attribute to force prop-dict to be present when some inherent attributes aren't present in the format (just a comment passing by, nothing blocking your work of course)